vocabulary-draft/continuous-integration-pipeline-events.md

```diff
@@ -0,0 +1,29 @@
+# Continuous Integration Pipeline Events
+
+These events are related to continuous integration pipelines, this pipelines usually include building, testing, packaging and releasing software artifacts. 
+Due the dynamic nature of Pipelines, most of actual work needs to be queued to happen in a distributed way, hence Queued events are added. 
+Adopters can choose to ignore these events if they don't apply to their use cases. 
+
+A pipeline, in the context of Continuous Integration, is the definition of a set of tasks that needs to be performed to build, test, package and release software artifacts. A pipeline can be instanciated multiple times, for example to build different versions of the same artifact. We are refering to this instance as PipelineRun. It will have a unique Id and it will help us to track the build and release progress on a particular software artifact. 
+
+- **PipelineRun Queued**: a PipelineRun has been schedule to run
+- **PipelineRun Started**: a PipelineRun has started and it is running
+- **PipelineRun Finished**: a PipelineRun has finished it execution, the event will contain the finished status, success, error or failure
+- **Build Queued**: a Build task has been queued, this build process usually is in charge of producing a binary from source code
+- **Build Started**: a Build task has started 
+- **Build Finished**: a Build task has finished, the event will contain the finished status, success, error or failure
```

Comment:
@m-linner-ericsson I do believe the same, but starting with a small set we can expand later. I do agree that adding Tasks, for example, is a good addition, also if we add an optional field for "Parent Pipeline" we can have a hierarchy. 
What's the difference between Step and Task? I know that Tekton uses `Task` and `TaskRun` to parallel `Pipeline` and `PipelineRun`

I am happy to include those events here.

Reply:
Or should we go with a general event instead example **Activity Queued/Triggered**, **Activity Started**, **Activity Finished**, and let the user define the type of activity inside the event?

Reply:
My comment from the old PR:
> We could consider adding a generic "activity" entity with queued/started/finished, but then Build and Tests may both be considered activities.

Reply:
@m-linner-ericsson I know that Tekton also has this low-level abstract. My opinion would be to keep them concrete to CD concepts. I do bet that each specific implementation will have this low-level of abstract concepts. The more abstract the concepts the more can be used for other purposes, defeating the purpose of defining the events related to CD specifically. I am conscious that this is my personal preference, I would love to hear opinions from @afrittoli and @e-backmark-ericsson

Reply:
I created a topic in Discussion to lift this discussion, as I believe it will have an impact on other areas of the spec as well https://github.com/cdfoundation/sig-events/discussions/15